static/js/dom.js

```diff
@@ -122,5 +124,18 @@ export let dom = {
             }
             dom.getBoardIdsFromDocument();
         }
+    },
+
+    renameBoard: function () {
+        let boardTitle = document.querySelectorAll('.board-title');
+        for (let title of boardTitle) {
+            title.addEventListener('dblclick', function (event) {
+                let original = event.currentTarget;
+                event.currentTarget.outerHTML = `<input type="text" class="new-board-title" value="${original.textContent}" size="15" required minlength="1" >` +
+                    `<button class="title-save-button">Save</button>`;
+                let saveButton = document.querySelector('.title-save-button');
+                saveButton.addEventListener('click', dataHandler.submitNewTitle);
```

Comment:
Here is the end of my review, I also tried to run the application and I tried the renaming function, it worked well.
Nice work, I had only little things / refactors to mention. Great PR! 😉